source/MaterialXGenShader/ShaderTranslator.h

```diff
@@ -0,0 +1,76 @@
+//
+// TM & (c) 2017 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#ifndef MATERIALX_SHADERTRANSLATOR_H
+#define MATERIALX_SHADERTRANSLATOR_H
+
+#include <MaterialXGenShader/Library.h>
+#include <MaterialXGenShader/Shader.h>
+#include <MaterialXGenShader/Util.h>
+
+namespace MaterialX
+{
+
+/// A shared pointer to a TextureBaker
+using ShaderTranslatorPtr = shared_ptr<class ShaderTranslator>;
+
+/// @class ShaderTranslator
+class ShaderTranslator
+{
+  public:
+    /// Translates shaderRef to the destShader shading model
+    void translateShader(ShaderRefPtr shaderRef, string destShader);
+
+    /// Translates all the materials to the destShader shading model if translation exists.
+    static void translateAllMaterials(DocumentPtr doc, string destShader);
+
+    /// Returns set of all the available potential translations
+    StringSet getAvailableTranslations(string start)
+    {
+        return _shadingTranslations[start];
+    }
+
+  protected:
+    /// Constructor.
+    static ShaderTranslatorPtr create(ConstDocumentPtr doc)
```

Comment:
Should this static creation method be public and accessible via Python?

Reply:
It wouldn't hurt! I will go ahead and change that.